Custom Branding

MatthiasReumann opened the pull request:
# Changes 
- Update Dockerfile
- Add API Route `/logo.svg` and `/manifest.json`
- Add `branding.go` which reads a `branding.yaml` file for the customization of titles, etc.
- Update README.md

# How it works 

Mount `branding` folder. Specify the folder in `config.yaml`. The custom logo and favicon should be visible.

If a `branding.yaml` is provided at the same location as `config.yaml`, it will be used for titles, etc.

If the `branding` folder doesn't exist the default TUM-Live configuration is used.

MatthiasReumann committed:
Update Dockerfile

Add copy of opptional branding folder for
- manifest.json
- favicon.ico
- logo.svg

MatthiasReumann committed:
Update README.md & Add /branding-example

Add Customization to README.md
Add branding-example folder

MatthiasReumann committed:
Add branding.go

Custom branding for titles, etc.
Add branding.yaml to /branding-example
Update templates

MatthiasReumann committed:
Add additional branding.yaml locations

Copied from config.go for config.yaml

MatthiasReumann committed:
Update README.md

Add branding.yaml information

MatthiasReumann committed:
Update docker-compose.yml

Add branding to volumes

MatthiasReumann committed:
Add custom description to branding.yaml

joschahenningsen commented on Dockerfile:
```diff
@@ -3,6 +3,11 @@ FROM node:18 as node
 WORKDIR /app
 COPY web web
 
+# Copy branding (if exists)
+COPY 'brandin[g]/manifest.json' /app/web/assets/
+COPY 'brandin[g]/favicon.ico' /app/web/assets/
+COPY 'brandin[g]/logo.svg' /app/web/assets/img/
```
Would it be possible to mount these with a volume like the branding.yaml?
End users should not be required to build the app themselves imo. 

Changes look great otherwise!

MatthiasReumann replied:
Good idea. Will look into it!

MatthiasReumann committed:
Merge branch 'dev' into enhancement/607-custom-branding

MatthiasReumann committed:
Add comments to functions

MatthiasReumann committed:
Remove copying of branding

MatthiasReumann committed:
Rename branding-example to branding

MatthiasReumann committed:
Update static router

Make static files more "dynamic"
Use embedded FS for favicon.ico
Include branding

MatthiasReumann committed:
go mod tidy

MatthiasReumann committed:
Update README.md

MatthiasReumann committed:
Merge branch 'dev' into enhancement/607-custom-branding

MatthiasReumann committed:
Add missing bg-white
